actionpack/lib/action_dispatch/http/mime_negotiation.rb

```diff
@@ -151,6 +154,17 @@ def valid_accept_header
       def use_accept_header
         !self.class.ignore_accept_header
       end
+
+      private
+
+      def extract_variant!
+        if parameters.key?(:format)
+          segments = parameters[:format].to_s.split("+")
```

Comment:
Very nice :+1:

I thought the same of you, but tests like below are not passed. Are these test invalid?

```ruby
test "can override format with parameter positive" do
  request = stub_request
  request.expects(:parameters).at_least_once.returns({ :format => :xml })
  assert request.format.xml?
end
```